django/db/models/sql/compiler.py

```diff
@@ -1891,7 +1895,13 @@ def contains_self_reference_subquery(self):
         )
 
     def _as_sql(self, query):
-        delete = "DELETE FROM %s" % self.quote_name_unless_alias(query.base_table)
+        result = [
+            "DELETE",
+            *self.query.comments,
```

Comment:
What's the rationale for injecting comments here instead of at the end of the query?

Reply:
Thanks for your comment. Queries can be long and truncated - please refer to the comment https://github.com/django/django/pull/15711#issuecomment-1132633387

Reply:
Yeah, the idea is that in a running query list, you can see the type of query first, then any tracing info. I saw this recommended by MySQL experts years ago, hence my suggestion for it in the previous PR. This is not for supporting any MySQL specific options.

I'm not so sure now that we should try to support inserting comments at any of the positions used for query hints. That would require a very complex API and might still not cover all cases. Also, database instrumentation exists to allow rewriting queries when needed.

Reply:
> I saw this recommended by MySQL experts years ago.

Do you have references for that? Isn't it fair to assume that an approach that works well for MySQL and recommended by these experts might not work so well for other backends (e.g. Postgres). I've seen plenty of [solutions](https://github.com/google/sqlcommenter/blob/c447d218a1b8ff628571b22cf4e74f2a6fe2e38a/python/sqlcommenter-python/google/cloud/sqlcommenter/__init__.py#L46-L53) and APMs default to prepending the traces to queries to ensure they are never truncated.

At work we built a solution that postpends commented traces specifically for MySQL and it appears to be working well hence my curiosity here.

Reply:
From an internal discussion with DBREs

1. Some tooling doesn’t support parsing in the middle of a query
2. Keeping queries starting with their respective verb makes it easier to filter
3. Putting the trace at the end of query might cause it to be truncated off on some systems

This makes me think that a proper solution would at the very least need to be configurable?